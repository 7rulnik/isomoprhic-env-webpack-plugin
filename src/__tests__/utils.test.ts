import { describe, expect, test } from '@jest/globals'

import {
	getClientEnvs,
	getServerEnvs,
	getGlobalVariable,
	getScriptTag,
} from '../'

describe('getGlobalVariable', () => {
	test('should use default variable if argument not passed', () => {
		expect(getGlobalVariable()).toBe('self.__ISOMORPHIC_ENV__')
	})

	test('should use variable from arguments', () => {
		expect(getGlobalVariable('__HELLO_WORLD__')).toBe('self.__HELLO_WORLD__')
	})
})

process.env.S_VAR_FOR_SERVER = 'VAR_FOR_SERVER'
process.env.C_VAR_FOR_CLIENT = 'VAR_FOR_CLIENT'
process.env.CS_VAR_FOR_CLIENT_AND_SERVER = 'VAR_FOR_CLIENT_AND_SERVER'
process.env.SC_VAR_FOR_SERVER_AND_CLIENT = 'VAR_FOR_SERVER_AND_CLIENT'

describe('getServerEnvs', () => {
	test('should return only variables for server', () => {
		expect(getServerEnvs()).toStrictEqual({
			VAR_FOR_SERVER: 'VAR_FOR_SERVER',
			VAR_FOR_CLIENT_AND_SERVER: 'VAR_FOR_CLIENT_AND_SERVER',
			VAR_FOR_SERVER_AND_CLIENT: 'VAR_FOR_SERVER_AND_CLIENT',
		})
	})
})

describe('getClientEnvs', () => {
	test('should return only variables for client', () => {
		expect(getClientEnvs()).toStrictEqual({
			VAR_FOR_CLIENT: 'VAR_FOR_CLIENT',
			VAR_FOR_CLIENT_AND_SERVER: 'VAR_FOR_CLIENT_AND_SERVER',
			VAR_FOR_SERVER_AND_CLIENT: 'VAR_FOR_SERVER_AND_CLIENT',
		})
	})
})

describe('getScriptTag', () => {
	test('should use default variable if argument not passed', () => {
		const result = getScriptTag()
		expect(result).toMatchSnapshot()
		expect(result).not.toMatch(/\"VAR_FOR_SERVER\"/)
	})

	test('should use variable from arguments', () => {
		const result = getScriptTag('__HELLO_WORLD__')
		expect(result).toMatchSnapshot()
		expect(result).not.toMatch(/\"VAR_FOR_SERVER\"/)
	})
})
