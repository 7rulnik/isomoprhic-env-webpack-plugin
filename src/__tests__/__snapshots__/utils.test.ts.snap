// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getScriptTag should use default variable if argument not passed 1`] = `
"<script>self.__ISOMORPHIC_ENV__ = {
  \\"VAR_FOR_CLIENT\\": \\"VAR_FOR_CLIENT\\",
  \\"VAR_FOR_CLIENT_AND_SERVER\\": \\"VAR_FOR_CLIENT_AND_SERVER\\",
  \\"VAR_FOR_SERVER_AND_CLIENT\\": \\"VAR_FOR_SERVER_AND_CLIENT\\"
}</script>"
`;

exports[`getScriptTag should use variable from arguments 1`] = `
"<script>self.__HELLO_WORLD__ = {
  \\"VAR_FOR_CLIENT\\": \\"VAR_FOR_CLIENT\\",
  \\"VAR_FOR_CLIENT_AND_SERVER\\": \\"VAR_FOR_CLIENT_AND_SERVER\\",
  \\"VAR_FOR_SERVER_AND_CLIENT\\": \\"VAR_FOR_SERVER_AND_CLIENT\\"
}</script>"
`;
